app/controllers/api/v1/likes_controller.rb

```diff
@@ -0,0 +1,69 @@
+class Api::V1::LikesController < ApplicationController
+  before_action :set_like, only: %i[ show update destroy ]
+  include Authentication
+
+  # GET /likes
+  def index
+    likes = Like.all
+
+    render json: likes
+  end
+
+  # GET /likes/1
+  def show
+    render json: like
+  end
+
+  # POST /likes
+  def create
+    if authenticate
+      post = Post.find(params[:post_id])
+      user = User.find_by(username: request.headers[:username])
+      like = Like.new(post: post, user: user)
+
+      unless Like.where(post_id: post.id, user_id: user.id).present?
+        if like.save
+          likes_count = post.likes_count + 1
+          post.update(likes_count: likes_count)
+
+          render json: {post: post}
+        else
+          render json: like.errors, status: :unprocessable_entity
+        end
+      else
+        render status: 400
+      end
+    else
+      render status: 401
+    end
+  end
+
+  # PATCH/PUT /likes/1
+  def update
+    if like.update(like_params)
```

Comment:
is it possible to change the post id for a particular like?

Reply:
No. In this case, should I `pass` function or should I return an error?

Reply:
No use case of this end-point

Reply:
Should I change the routing from using resources to manual routing? Because if I remove the method then sending patch request breaks the code